CONTRIBUTING.md

```diff
@@ -0,0 +1,27 @@
+# Contributing
+This document describes how to go about contributing to the ISOBlue-Avena repository as well as defining how we manage pull requests, releases, and other collaborative features
+
+
+## General Guidelines
+If you are looking to help us, please visit our [issue tracker](https://github.com/OATS-Group/isoblue-avena/issues). If you would like to suggest or discuss a feature/bug, please email our mailing list found here https://groups.google.com/forum/?oldui=1#!forum/isoblue. Please ensure you mention the 'Avena' version of ISOBlue.
+
+
+## Release Branches and Versioning
+The master branch will track the current unstable version of the codebase (equivalent to the `dev` branch in many other repositories). Stable versions will be maintained using branches in the form of `release/xx.yy.zz`, similar to the [Gitlab Flow](https://docs.gitlab.com/ee/topics/gitlab_flow.html) style. This is primarily for the ability to cherry-pick bug fixes into previous releases. 
+
+Releases will be numbered in the [Semantic Versioning](https://semver.org/) style. Version `XX.YY.ZZ` is major version `XX`, minor version `YY`, and bugfix `ZZ`. Major versions incorporate breaking changes, minor versions introduce new features, and bugfixes... fix bugs
+
+
+## Feature Branches and Pull Requests
+Members of the ISOBlue-Avena repository will use feature/bugfix branches in the form of `feature/feature-name` or `bugfix/bug-name`. Outside contributors forking the repository are encouraged to do the same
+
+
+Once a feature/bugfix branch as reached maturity and has been properly [tested](#Testing) a pull request will be created requesting to merge the branch into master. Members of the ISOBlue-Avena repository will review the code and changes may be be requested. Given all requested changes are addressed and all tests pass, a member of the repository will approve the changes and the branch will be merged into master. If the person opening the pull request has permission to merge, it will be their responsibilty to merge the branch and reslove any conflicts. Otherwise this responsibility falls to the first reviewer listed in the Github Pull Request UI
```

Comment:
>If the person opening the pull request has permission to merge, it will be their responsibility to merge the branch and resolve any conflicts. Otherwise this responsibility falls to the first reviewer listed in the Github Pull Request UI

It seems we get into these limbo states where a PR is ready to merge but none of us merge it. I think having someone who is responsible for 'pushing the button' by default might help. Unsure how to assign this person so if there is a less clunky way to assign them more than happy to amend

Reply:
I think assigning a deadline would be useful in these situations. If the PR is ready to merge (with this I mean, no conflicts) and no further discussion has been done in 72 hours, the project "manager" would "push the button".

Reply:
I think that could be pretty dangerous.

Something like simple updates to a single service where the context is well understood, maybe it is okay. However, something like changes to the underlying system may have enough impact that it could take a person a while to find the time to testing / evaluate.

Reply:
Agree. Instead of a deadline... What about after the PR is approved, the author of the PR would be the one resposible of pushing the button. I mean, assuming that the approval came after thoroughly testing the code changes.

Reply:
@facastiblancor I agree. That is why I initially said the person who opened it should push the button. However I believe that you can open a PR without having permission to merge it into master (think contributors from outside OATS-Group). Maybe we say they should ask one of the maintains to merge?

Reply:
@aaron97neu I agree. I think one of us just takes ownership of an outside PR. I'd suggest that  the first to triage the PR request reviews from some / all of us. In the process, we ask one to take leadership of it. We can use the PR comments to accept or attempt to push on someone else. Likely there will be a clear "owner" based on what is being changed.

As for pushing along stale PRs ... I think pinging usually works. Like today, @aaron97neu's request to merge the PR prompted me to squeeze out a few minutes to not be the bottle neck (which wasn't a problem, but the PR wasn't on the top of my mind to think about doing it). Also, the IB team should have a weekly (?) standing meeting. Could be fast most weeks, but that's a time to push on stale things.

Reply:
Modified include @abalmos's point

>If the person opening the pull request has permission to merge, it will be their responsibility to merge the branch and resolve any conflicts, otherwise one of the repository maintainers 

Considered mentioned it was also their responsibility to push along the PR if it gets stale and stuff but could not get it to flow well and felt like it was kinda implicit anyways


I think having regular IB team meetings is very important. If they are standup-style (>10mins iirc)I would even go for more often that.

Reply:
If you mean < 10 mins, then I concur. We have a few milestones to hit in a shortish period 

This PR still LGTM